### PR TITLE
Adds KCP local development script for managed-gitops (Solves: #GITOPSRVCE-158)

### PR DESCRIPTION
**Pull request by samyak-jn**
#### Description:
- This PR adds a local development shell script that setups KCP as a minimal api-server which in turn is used for development purposes. We are using release-0.5 which is a stable latest release of KCP to run our gitops service and test our changes on the same.
- This script presumes that the developer is already logged into an openshift cluster, such as a clusterbot cluster
- Note: This is in WIP, do not use it in production :P
- Note: Once, the script is ready I'll open it for the reviewers to review the same.

### CODE REVIEW

File: utilities/scripts/kcp/setup-kcp-local.sh [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    printf "\nThe KCP binaries are installed successfully ...\n"
+  fi
+}
+
+# Initiating the KCP server, the kubectl in this case will be using the KCP's kubeconfig
+kcp-init() {
+  printf "\nInitiating KCP server ...\n"
+  pushd "${TMP_DIR}"
+  pwd
+  "$($KCP_DIR/bin/kcp start)" &> "${TMP_DIR}/kcp.log" &
```
Comment (chetan-rns) [inline utilities/scripts/kcp/setup-kcp-local.sh:65]: Here the KCP process runs in the background on our local machine. For [GITOPSRVCE-158](https://issues.redhat.com/browse/GITOPSRVCE-158), shouldn't the KCP run in an actual physical cluster as a Deployment/Statefulset?

Comment (samyak-jn) [reply utilities/scripts/kcp/setup-kcp-local.sh:65]: From my understanding, the expectation from this story is to set up KCP as the api-server (logical cluster) and we are preassuming that we have logged into an openshift cluster that will be acting as our workload aka physical cluster.
Does that make sense to you?

Comment (chetan-rns) [reply utilities/scripts/kcp/setup-kcp-local.sh:65]: The story also talks about setting up the workspace on a physical cluster i.e to deploy KCP as a Deployment/StatefulSet.  

```
Setup kcp workspace on physical cluster
    * deploy kcp on cluster (in a Deployment/StatefulSet) with the appropriate settings for us
    * deploy a syncer on cluster (in a Deployment/StatefulSet) with appropriate settings for us
```

I might be wrong. @jgwest For this story, do we expect KCP to run on our local machine or as a container in an OCP cluster?

File: utilities/scripts/kcp/manifests/workloadcluster.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+apiVersion: workload.kcp.dev/v1alpha1
+kind: WorkloadCluster
+metadata:
+ name: local
+spec:
+ kubeconfig: |
```
Comment (chetan-rns) [inline utilities/scripts/kcp/manifests/workloadcluster.yaml:6]: I was checking the WorkloadCluster API and looks like it no longer has the `kubeconfig` field. When I sync a workload using the `kubectl kcp workload sync` cmd the WorkloadCluster resource is created automatically. So maybe we don't have to manually create this workload resource.

https://github.com/kcp-dev/kcp/blob/f58e582a75b53d6bc5c9ceb4551f18e40f0f240a/pkg/apis/workload/v1alpha1/workloadcluster_types.go#L56

**Review by jgwest — CHANGES_REQUESTED** [review]
@samyak-jn I'm seeing these errors when running the script:

```
* Initializing DB
  Copying db-schema.sql into the postgres container.
cd /tmp/test/managed-gitops/utilities/db-migration && go run main.go
Unable to migrate database: SEVERE: migration could not be applied; no migration found for version 4: read down for version 4 .: file does not exist
exit status 1
make: *** [Makefile:237: db-migrate] Error 1
```
	
```	
07:27:55              backend | 2022-07-20T07:27:55.643747135-04:00	ERROR	setup	Fatal Error: Unsuccessful Migration	{"error": "SEVERE: migration could not be applied; no migration found for version 4: read down for version 4 .: file does not exist"}
07:27:55              backend | main.main
07:27:55              backend | 	/tmp/test/managed-gitops/backend/main.go:96
07:27:55              backend | runtime.main
07:27:55              backend | 	/usr/local/go/src/runtime/proc.go:250
07:27:55              backend | exit status 1
07:27:55              backend | make[1]: *** [Makefile:60: run] Error 1
07:27:55              backend | make[1]: Leaving directory '/tmp/test/managed-gitops/backend'
07:27:55              backend | Terminating backend
```

Anything special I need to do to run it? This is what I am currently doing:
```
# ensure kubectl-kcp is on my path
git clone https://github.com/samyak-jn/managed-gitops
cd managed-gitops
git checkout add-kcp-local-script 
utilities/scripts/kcp/setup-kcp-local.sh
```
(I also tried running `setup-kcp-local.sh` from inside the scripts directory, but hit a similar but different issue)

**Comment by samyak-jn** [comment]
@jgwest, sorry for the late reply, I'll try to reproduce this error at my end, not sure why it is happening atm, will debug it and let you know. I and Chetan have tried to run this script and haven't encountered this issue.

**Comment by samyak-jn** [comment]
So, by exactly following the steps that you mentioned

>
> #ensure kubectl-kcp is on my path
> git clone https://github.com/samyak-jn/managed-gitops
> cd managed-gitops
> git checkout add-kcp-local-script 
> ./utilities/scripts/kcp/setup-kcp-local.sh
> 

Here are the logs that I'm having: https://paste.debian.net/hidden/a9397626/ (full logs here)
The E2E tests are failing due to a timeout error, which IMHO is a KCP issue, but the one you are facing is not that I can reproduce.

Error on E2E:

```
=== RUN   TestCore
Running Suite: Core Suite - /home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core
=====================================================================================================
Random Seed: 1658820881

Will run 4 of 4 specs
I0726 13:04:44.806517   48406 placement_controller.go:263] Queueing Namespace root:default:dev-work|gitops-service-e2e
I0726 13:04:44.806548   48406 namespace_reconcile.go:45] Reconciling namespace root:default:dev-work|gitops-service-e2e
I0726 13:04:44.807965   48406 resource_reconcile.go:48] Reconciling GVR "/v1, Resource=serviceaccounts" root:default:dev-work|gitops-service-e2e/default
I0726 13:04:44.808122   48406 resource_reconcile.go:48] Reconciling GVR "/v1, Resource=configmaps" root:default:dev-work|gitops-service-e2e/kube-root-ca.crt
I0726 13:04:44.808770   48406 placement_controller.go:263] Queueing Namespace root:default:dev-work|gitops-service-e2e
I0726 13:04:44.808825   48406 namespace_reconcile.go:45] Reconciling namespace root:default:dev-work|gitops-service-e2e
I0726 13:04:44.808863   48406 placement_reconcile_status.go:74] Updating status for namespace root:default:dev-work|gitops-service-e2e: {"metadata":{"resourceVersion":"310","uid":"f85293db-bf9e-451b-aa65-f7f95133371b"},"status":{"conditions":[{"lastTransitionTime":"2022-07-26T07:34:44Z","message":"No workload clusters available under given placement constraints","reason":"Unschedulable","status":"False","type":"NamespaceScheduled"}]}}
I0726 13:04:44.810737   48406 placement_controller.go:263] Queueing Namespace root:default:dev-work|gitops-service-e2e
I0726 13:04:44.810768   48406 namespace_reconcile.go:45] Reconciling namespace root:default:dev-work|gitops-service-e2e
I0726 13:04:44.819884   48406 resource_reconcile.go:48] Reconciling GVR "/v1, Resource=secrets" root:default:dev-work|gitops-service-e2e/default-token-nwqw5
I0726 13:04:44.821496   48406 resource_reconcile.go:48] Reconciling GVR "/v1, Resource=serviceaccounts" root:default:dev-work|gitops-service-e2e/default
I0726 13:05:25.235986   48406 informer.go:253] Discovering types for logical cluster "root:default"
I0726 13:05:25.238438   48406 informer.go:253] Discovering types for logical cluster "root:default:dev-work"
I0726 13:06:25.235634   48406 informer.go:253] Discovering types for logical cluster "root:default"
I0726 13:06:25.239368   48406 informer.go:253] Discovering types for logical cluster "root:default:dev-work"
------------------------------
• [FAILED] [123.174 seconds]
GitOpsDeployment Status Tests
/home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core/gitops_deployment_status_test.go:12
  Status field of GitOpsDeployment is updated accurately
  /home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core/gitops_deployment_status_test.go:13
    [It] GitOpsDeployment .status.resources field is populated with the right resources
    /home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core/gitops_deployment_status_test.go:14

  Expected success, but got an error:
      <*errors.errorString | 0xc0003e2000>: {
          s: "timed out waiting for the condition",
      }
      timed out waiting for the condition
  In [It] at: /home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core/gitops_deployment_status_test.go:15
------------------------------
I0726 13:06:46.892131   48406 placement_controller.go:263] Queueing Namespace root:default:dev-work|gitops-service-e2e
I0726 13:06:46.892148   48406 namespace_reconcile.go:45] Reconciling namespace root:default:dev-work|gitops-service-e2e
I0726 13:06:51.913038   48406 store.go:1019] DEBUG: DELETE key func returned: /core/configmaps/root:default:dev-work/gitops-service-e2e/kube-root-ca.crt
I0726 13:06:51.986152   48406 store.go:1487] "Monitoring resource count at path" resource="components.appstudio.redhat.com" path="<storage-prefix>//appstudio.redhat.com/components/customresources"
I0726 13:06:51.991727   48406 store.go:1019] DEBUG: DELETE key func returned: /core/secrets/root:default:dev-work/gitops-service-e2e/default-token-nwqw5
I0726 13:06:51.996639   48406 resource_reconcile.go:48] Reconciling GVR "/v1, Resource=serviceaccounts" root:default:dev-work|gitops-service-e2e/default
I0726 13:06:52.001756   48406 store.go:1019] DEBUG: DELETE key func returned: /core/serviceaccounts/root:default:dev-work/gitops-service-e2e/default
E0726 13:06:52.008829   48406 tokens_controller.go:266] error synchronizing serviceaccount root:default:dev-work|gitops-service-e2e/default: secrets "default-token-5w7jg" is forbidden: unable to create new content in namespace gitops-service-e2e because it is being terminated
I0726 13:06:52.010433   48406 placement_controller.go:263] Queueing Namespace root:default:dev-work|gitops-service-e2e
I0726 13:06:52.010460   48406 namespace_reconcile.go:45] Reconciling namespace root:default:dev-work|gitops-service-e2e
I0726 13:06:52.010583   48406 placement_reconcile_status.go:74] Updating status for namespace root:default:dev-work|gitops-service-e2e: {"metadata":{"resourceVersion":"319","uid":"f85293db-bf9e-451b-aa65-f7f95133371b"},"status":{"conditions":[{"lastTransitionTime":"2022-07-26T07:36:52Z","message":"All content successfully removed","reason":"ContentRemoved","status":"False","type":"NamespaceContentRemaining"},{"lastTransitionTime":"2022-07-26T07:36:52Z","message":"All content successfully deleted, may be waiting on finalization","reason":"ContentDeleted","status":"False","type":"NamespaceDeletionContentFailure"},{"lastTransitionTime":"2022-07-26T07:36:52Z","message":"All resources successfully discovered","reason":"ResourcesDiscovered","status":"False","type":"NamespaceDeletionDiscoveryFailure"},{"lastTransitionTime":"2022-07-26T07:36:52Z","message":"All legacy kube types successfully parsed","reason":"ParsedGroupVersions","status":"False","type":"NamespaceDeletionGroupVersionParsingFailure"},{"lastTransitionTime":"2022-07-26T07:36:52Z","message":"All content-preserving finalizers finished","reason":"ContentHasNoFinalizers","status":"False","type":"NamespaceFinalizersRemaining"},{"lastTransitionTime":"2022-07-26T07:34:44Z","message":"No workload clusters available under given placement constraints","reason":"Unschedulable","status":"False","type":"NamespaceScheduled"}]}}
I0726 13:06:52.013372   48406 placement_controller.go:263] Queueing Namespace root:default:dev-work|gitops-service-e2e
I0726 13:06:52.013383   48406 namespace_reconcile.go:45] Reconciling namespace root:default:dev-work|gitops-service-e2e
I0726 13:06:52.018135   48406 placement_controller.go:263] Queueing Namespace root:default:dev-work|gitops-service-e2e
I0726 13:06:52.018157   48406 namespace_reconcile.go:45] Reconciling namespace root:default:dev-work|gitops-service-e2e
I0726 13:07:25.235362   48406 informer.go:253] Discovering types for logical cluster "root:default"
I0726 13:07:25.237687   48406 informer.go:253] Discovering types for logical cluster "root:default:dev-work"
I0726 13:08:25.235946   48406 informer.go:253] Discovering types for logical cluster "root:default"
I0726 13:08:25.239200   48406 informer.go:253] Discovering types for logical cluster "root:default:dev-work"
• [FAILED] [122.081 seconds]
GitOpsDeployment E2E tests
/home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core/simple_test.go:14
  Create a new GitOpsDeployment
  /home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core/simple_test.go:16
    [It] should be healthy and have synced status, and resources should be deployed
    /home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core/simple_test.go:18

  Expected success, but got an error:
      <*errors.errorString | 0xc0003e2000>: {
          s: "timed out waiting for the condition",
      }
      timed out waiting for the condition
  In [It] at: /home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core/simple_test.go:20
------------------------------
I0726 13:08:49.023681   48406 store.go:1019] DEBUG: DELETE key func returned: /core/namespaces/root:default:dev-work/gitops-service-e2e
I0726 13:08:49.025558   48406 placement_controller.go:263] Queueing Namespace root:default:dev-work|gitops-service-e2e
I0726 13:08:49.025582   48406 namespace_reconcile.go:45] Reconciling namespace root:default:dev-work|gitops-service-e2e
I0726 13:09:25.235276   48406 informer.go:253] Discovering types for logical cluster "root:default"
I0726 13:09:25.237709   48406 informer.go:253] Discovering types for logical cluster "root:default:dev-work"
W0726 13:09:54.262960   48406 watcher.go:245] watch chan error: etcdserver: mvcc: required revision has been compacted
I0726 13:10:25.235241   48406 informer.go:253] Discovering types for logical cluster "root:default"
I0726 13:10:25.238062   48406 informer.go:253] Discovering types for logical cluster "root:default:dev-work"
• [FAILED] [122.138 seconds]
Standalone ArgoCD instance E2E tests
/home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core/argocd_instance_test.go:21
  Create a Standalone ArgoCD instance [BeforeEach]
  /home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core/argocd_instance_test.go:31
    should create ArgoCD resource and application, wait for it to be installed and synced
    /home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core/argocd_instance_test.go:41

  Begin Captured GinkgoWriter Output >>
    STEP: Delete old namespaces, and kube-system resources 07/26/22 13:08:46.906
  << End Captured GinkgoWriter Output

  Expected success, but got an error:
      <*errors.errorString | 0xc0003e2000>: {
          s: "timed out waiting for the condition",
      }
      timed out waiting for the condition
  In [BeforeEach] at: /home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core/argocd_instance_test.go:34
------------------------------
I0726 13:10:51.114488   48406 store.go:1019] DEBUG: DELETE key func returned: /core/namespaces/root:default:dev-work/gitops-service-e2e
13:11:19              backend | 2022-07-26T13:11:19.360570901+05:30	INFO	task retry loop status [event-loop-router-retry-loop]: waitingTasks: 0, activeTasks: 0 	{"task-retry-name": "event-loop-router-retry-loop"}
I0726 13:11:25.235350   48406 informer.go:253] Discovering types for logical cluster "root:default"
I0726 13:11:25.246239   48406 informer.go:253] Discovering types for logical cluster "root:default:dev-work"
13:12:02        cluster-agent | 2022-07-26T13:12:02.93238271+05:30	INFO	task retry loop status [cluster-agent]: waitingTasks: 0, activeTasks: 0 	{"task-retry-name": "cluster-agent"}
13:12:02        cluster-agent | 2022-07-26T13:12:02.934610135+05:30	INFO	task retry loop status [application-reconciler]: waitingTasks: 0, activeTasks: 0 	{"task-retry-name": "application-reconciler"}
I0726 13:12:25.235321   48406 informer.go:253] Discovering types for logical cluster "root:default"
I0726 13:12:25.237796   48406 informer.go:253] Discovering types for logical cluster "root:default:dev-work"
• [FAILED] [122.079 seconds]
Argo CD Application
/home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core/argocd_application_test.go:20
  Creating GitOpsDeployment should result in an Argo CD Application
  /home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core/argocd_application_test.go:21
    [It] Argo CD Application should have has prune, allowEmpty and selfHeal enabled
    /home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core/argocd_application_test.go:22

  Expected success, but got an error:
      <*errors.errorString | 0xc0003e2000>: {
          s: "timed out waiting for the condition",
      }
      timed out waiting for the condition
  In [It] at: /home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core/argocd_application_test.go:23
------------------------------


Summarizing 4 Failures:
  [FAIL] GitOpsDeployment Status Tests Status field of GitOpsDeployment is updated accurately [It] GitOpsDeployment .status.resources field is populated with the right resources
  /home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core/gitops_deployment_status_test.go:15
  [FAIL] GitOpsDeployment E2E tests Create a new GitOpsDeployment [It] should be healthy and have synced status, and resources should be deployed
  /home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core/simple_test.go:20
  [FAIL] Standalone ArgoCD instance E2E tests Create a Standalone ArgoCD instance [BeforeEach] should create ArgoCD resource and application, wait for it to be installed and synced
  /home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core/argocd_instance_test.go:34
  [FAIL] Argo CD Application Creating GitOpsDeployment should result in an Argo CD Application [It] Argo CD Application should have has prune, allowEmpty and selfHeal enabled
  /home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e/core/argocd_application_test.go:23

Ran 4 of 4 Specs in 489.477 seconds
FAIL! -- 0 Passed | 4 Failed | 0 Pending | 0 Skipped
--- FAIL: TestCore (489.48s)
FAIL
coverage: [no statements]
FAIL	github.com/redhat-appstudio/managed-gitops/tests-e2e/core	489.592s
?   	github.com/redhat-appstudio/managed-gitops/tests-e2e/fixture	[no test files]
?   	github.com/redhat-appstudio/managed-gitops/tests-e2e/fixture/application	[no test files]
?   	github.com/redhat-appstudio/managed-gitops/tests-e2e/fixture/gitopsdeployment	[no test files]
?   	github.com/redhat-appstudio/managed-gitops/tests-e2e/fixture/k8s	[no test files]
FAIL
make[1]: *** [Makefile:36: test] Error 1
make[1]: Leaving directory '/home/samjain/gitops-service-team/kc-script/managed-gitops/tests-e2e'
```

**Comment by jgwest** [comment]
It is now working for me, thanks for investigating @samyak-jn! I continue to review as of this evening.

**Comment by jgwest** [comment]
@samyak-jn , looks like I'm hitting a new issue :thinking::

### Steps to reproduce:

```
# Make sure no old kcp are running:
killall kcp
killall kubectl
killall go

# Erase my old kube config, to make sure no other settings are interfering
rm ~/.kube/config

# Re-login to OpenShift clusterbot cluster

mkdir /tmp/test
cd /tmp/test
git clone https://github.com/samyak-jn/managed-gitops
cd managed-gitops
git checkout add-kcp-local-script
./utilities/scripts/kcp/setup-kcp-local.sh
```

After a few moments, I see this when I look on the workload cluster, in the `kcpsync*` namespace. The `kcp-syncer` pod is failing:
```
[jgw@localhost-lan managed-gitops]$ kubens | grep kcp
kcpsync1bce75f6e80f02feb8bb20bab4da3a0eae78f8b13f1c86b2f51571ab
[jgw@localhost-lan managed-gitops]$ k get all -n kcpsync1bce75f6e80f02feb8bb20bab4da3a0eae78f8b13f1c86b2f51571ab
NAME                              READY   STATUS    RESTARTS       AGE
pod/kcp-syncer-5fdfd5486d-rcbft   1/1     Running   5 (100s ago)   5m42s

NAME                         READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/kcp-syncer   1/1     1            1           5m42s

NAME                                    DESIRED   CURRENT   READY   AGE
replicaset.apps/kcp-syncer-5fdfd5486d   1         1         1       5m42s

Here are the logs from the pod:
```
[jgw@localhost-lan managed-gitops]$ k  logs -n kcpsync1bce75f6e80f02feb8bb20bab4da3a0eae78f8b13f1c86b2f51571ab pod/kcp-syncer-5fdfd5486d-rcbft 
I0726 23:12:10.913584       1 syncer.go:70] Syncing the following resource types: [clusterrolebindings clusterroles configmaps deployments.apps networkpolicies pods rolebindings roles secrets serviceaccounts services statefulsets]
I0726 23:12:10.914611       1 syncer.go:77] Starting syncer for logical-cluster: root:default:dev-work, workload-cluster: api-ci-ln-k9s3cyb-72292-origin-ci-int-gce-dev-rhcloud-com
I0726 23:12:10.914776       1 syncer.go:94] Attempting to retrieve the Syncer virtual workspace URL from WorkloadCluster root:default:dev-work|api-ci-ln-k9s3cyb-72292-origin-ci-int-gce-dev-rhcloud-com
E0726 23:12:40.916079       1 run.go:74] "command failed" err="Get \"https://192.168.8.150:6443/clusters/root:default:dev-work/apis/workload.kcp.dev/v1alpha1/workloadclusters/api-ci-ln-k9s3cyb-72292-origin-ci-int-gce-dev-rhcloud-com\": dial tcp 192.168.8.150:6443: i/o timeout"
```

It looks like it's trying to connect to an IP address that isn't likely to exist on the cluster.

The IP address that it's attempting to connect to seems to come from the generated `syncer.yaml`:
```
[jgw@localhost-lan managed-gitops]$ cat /tmp/kcp-gitops-service.a8saKRUb8/syncer.yaml 
---
apiVersion: v1
kind: Namespace
metadata:
  name: kcpsync1bce75f6e80f02feb8bb20bab4da3a0eae78f8b13f1c86b2f51571ab
  labels:
    workload.kcp.io/logical-cluster: root_default_dev-work
    workload.kcp.io/workload-cluster: api-ci-ln-k9s3cyb-72292-origin-ci-int-gce-dev-rhcloud-com
---
apiVersion: v1
kind: ServiceAccount
metadata:
  name: kcp-syncer
  namespace:  kcpsync1bce75f6e80f02feb8bb20bab4da3a0eae78f8b13f1c86b2f51571ab
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: kcpsync1bce75f6e80f02feb8bb20bab4da3a0eae78f8b13f1c86b2f51571ab
rules:
- apiGroups:
  - ""
  resources:
  - namespaces
  verbs:
  - "create"
  - "list"
  - "watch"
- apiGroups:
  - "apiextensions.k8s.io"
  resources:
  - customresourcedefinitions
  verbs:
  - "get"
  - "watch"
  - "list"
- apiGroups:
  - ""
  resources:
  - clusterrolebindings
  - clusterroles
  - configmaps
  - networkpolicies
  - pods
  - rolebindings
  - roles
  - secrets
  - serviceaccounts
  - services
  - statefulsets
  verbs:
  - "*"
- apiGroups:
  - "apps"
  resources:
  - deployments
  verbs:
  - "*"
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: kcpsync1bce75f6e80f02feb8bb20bab4da3a0eae78f8b13f1c86b2f51571ab
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: kcpsync1bce75f6e80f02feb8bb20bab4da3a0eae78f8b13f1c86b2f51571ab
subjects:
- kind: ServiceAccount
  name: kcp-syncer
  namespace:  kcpsync1bce75f6e80f02feb8bb20bab4da3a0eae78f8b13f1c86b2f51571ab
---
apiVersion: v1
kind: Secret
metadata:
  name: kcp-syncer-config
  namespace:  kcpsync1bce75f6e80f02feb8bb20bab4da3a0eae78f8b13f1c86b2f51571ab
stringData:
  kubeconfig: |
    apiVersion: v1
    kind: Config
    clusters:
    - name: default-cluster
      cluster:
        certificate-authority-data: (...)
        server: https://192.168.8.150:6443
    contexts:
    - name: default-context
      context:
        cluster: default-cluster
        namespace: default
        user: default-user
    current-context: default-context
    users:
    - name: default-user
      user:
        token: (...)
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: kcp-syncer
  namespace:  kcpsync1bce75f6e80f02feb8bb20bab4da3a0eae78f8b13f1c86b2f51571ab
spec:
  replicas: 1
  strategy:
    type: Recreate
  selector:
    matchLabels:
      app: kcpsync1bce75f6e80f02feb8bb20bab4da3a0eae78f8b13f1c86b2f51571ab
  template:
    metadata:
      labels:
        app: kcpsync1bce75f6e80f02feb8bb20bab4da3a0eae78f8b13f1c86b2f51571ab
    spec:
      containers:
      - name: kcp-syncer
        command:
        - /ko-app/syncer
        args:
        - --from-kubeconfig=/kcp/kubeconfig
        - --workload-cluster-name=api-ci-ln-k9s3cyb-72292-origin-ci-int-gce-dev-rhcloud-com
        - --from-cluster=root:default:dev-work
        - --resources=clusterrolebindings
        - --resources=clusterroles
        - --resources=configmaps
        - --resources=deployments.apps
        - --resources=networkpolicies
        - --resources=pods
        - --resources=rolebindings
        - --resources=roles
        - --resources=secrets
        - --resources=serviceaccounts
        - --resources=services
        - --resources=statefulsets
        image: ghcr.io/kcp-dev/kcp/syncer:v0.5.0-alpha.1
        imagePullPolicy: IfNotPresent
        terminationMessagePolicy: FallbackToLogsOnError
        volumeMounts:
        - name: kcp-config
          mountPath: /kcp/
          readOnly: true
      serviceAccountName: kcp-syncer
      volumes:
        - name: kcp-config
          secret:
            secretName: kcp-syncer-config
            optional: false
[jgw@localhost-lan managed-gitops]$ 
```


My `.kube/config`, FWIW:
```yaml
[jgw@localhost-lan managed-gitops]$ cat ~/.kube/config
apiVersion: v1
clusters:
- cluster:
    insecure-skip-tls-verify: true
    server: https://api.ci-ln-k9s3cyb-72292.origin-ci-int-gce.dev.rhcloud.com:6443
  name: api-ci-ln-k9s3cyb-72292-origin-ci-int-gce-dev-rhcloud-com:6443
contexts:
- context:
    cluster: api-ci-ln-k9s3cyb-72292-origin-ci-int-gce-dev-rhcloud-com:6443
    namespace: default
    user: kube:admin/api-ci-ln-k9s3cyb-72292-origin-ci-int-gce-dev-rhcloud-com:6443
  name: default/api-ci-ln-k9s3cyb-72292-origin-ci-int-gce-dev-rhcloud-com:6443/kube:admin
current-context: default/api-ci-ln-k9s3cyb-72292-origin-ci-int-gce-dev-rhcloud-com:6443/kube:admin
kind: Config
preferences: {}
users:
- name: kube:admin/api-ci-ln-k9s3cyb-72292-origin-ci-int-gce-dev-rhcloud-com:6443
  user:
    token: sha256~(...)
```

**Comment by chetan-rns** [comment]
@samyak-jn  Like I mentioned yesterday, the timeout error here https://github.com/redhat-appstudio/managed-gitops/pull/170#issuecomment-1195127505 is because of missing roles and role bindings in the `gitops-service-e2e` namespace. In the function [EnsureCleanState()](https://github.com/redhat-appstudio/managed-gitops/blob/e9a8166b3ec903e946ba570c771511453728ca9d/tests-e2e/fixture/fixture.go#L240), we add the `managed-by` label and expect specific RBAC resources to be automatically created for us in the gitops-service-e2e namespace. This `managed-by` label feature is specific to the operator and doesn't work when installing argocd [manually](https://github.com/redhat-appstudio/managed-gitops/blob/2f1205130728d7c2b3a6407c31355913eee11a5a/utilities/scripts/kcp/setup-kcp-local.sh#L139).

So we should either add the missing RBAC manually from the script or figure out how to install argocd from the operator.

**Comment by samyak-jn** [comment]
The https://github.com/redhat-appstudio/managed-gitops/pull/200 has been merged against this issue/story, hence closing this! FYI, this PR can be tested on the local environment for testing (not e2e) which needs local k3d minikube etc, plus we have kcp-release 0.7 so might need to update the images as well.

**Pull request closed without merging**